ouroboros-consensus-shelley/src/Ouroboros/Consensus/Shelley/Ledger/Query.hs

```diff
@@ -575,3 +662,50 @@ decodeShelleyResult query = case query of
     GetStakePools                              -> fromCBOR
     GetStakePoolParams {}                      -> fromCBOR
     GetRewardInfoPools                         -> fromCBOR
+    GetPoolState {}                            -> fromCBOR
+    GetStakeSnapshot {}                        -> fromCBOR
+
+data StakeSnapshot crypto = StakeSnapshot
+  { sMarkPool   :: !Integer
+  , sSetPool    :: !Integer
+  , sGoPool     :: !Integer
+  , sMarkTotal  :: !Integer
+  , sSetTotal   :: !Integer
+  , sGoTotal    :: !Integer
+  } deriving (Eq, Show, Generic)
```

Comment:
Where should this type go?

Reply:
It's just used for a query result, that's all, right? In that case, I think this module is a reasonable place for it; since this module defines the Shelley queries -- the result type is effectively part of the query.

Reply:
Ok, keeping here.